### PR TITLE
Remove strformat.unicode_safe()

### PR DESCRIPTION
**Pull request by cjmayo**
I don't have the energy to check each use individually - this is following the rip it out and fix the tests approach.

What do you think?

### CODE REVIEW

**Comment by cjmayo** [comment]
Got a bit more rigorous about this and checked we actually had test coverage, two untested in plugins.locationinfo.get_location() 

https://github.com/linkchecker/linkchecker/blob/2a46b91327e7286edbf5a93d5fa71838033e630f/linkcheck/plugins/locationinfo.py#L114-L118

added a test. Unfortunately geoip-database does not have the city information so only one of these is tested but I think it is reasonable to assume the other would be OK.

File: linkcheck/plugins/parsepdf.py [hunk 1]
```diff
@@ -44,14 +44,7 @@ def search_url(obj, url_data, pageno, seen_objs):
...
             if key == 'URI':
-                if isinstance(value, str):
-                    url = value
-                else:
-                    # URIs should be 7bit ASCII encoded, but be safe and encode
-                    # to unicode
-                    # XXX this does not use an optional specified base URL
-                    url = strformat.unicode_safe(value)
-                url_data.add_url(url, page=pageno)
+                url_data.add_url(str(value, encoding="ascii"), page=pageno)
```
Comment (mgedmin) [inline linkcheck/plugins/parsepdf.py:47]: Whoa, I don't think I've ever seen `str(something, encoding='ascii')` before.  `something.decode('ascii')` is the canonical spelling of this operation.

File: linkcheck/checker/httpurl.py [hunk 2]
```diff
@@ -111,7 +110,7 @@ def add_size_info(self):
             # Note that content-encoding causes size differences since
             # the content data is always decoded.
             try:
-                self.size = int(self.getheader("Content-Length"))
+                self.size = int(self.headers.get("Content-Length"))
             except (ValueError, OverflowError):
```
Comment (mgedmin) [inline linkcheck/checker/httpurl.py:114]: int(None) will raise a TypeError.

I don't think you can get an OverflowError from int() in Python 3.

(Preexisting bug, fix, if any, belongs to a separate PR.)

File: linkcheck/checker/httpurl.py [hunk 3]
```diff
@@ -273,7 +272,7 @@ def follow_redirections(self, request):
             # Reset extern and recalculate
             self.extern = None
             self.set_extern(newurl)
-            self.urlparts = strformat.url_unicode_split(newurl)
+            self.urlparts = list(urllib.parse.urlsplit(newurl))
```
Comment (mgedmin) [inline linkcheck/checker/httpurl.py:275]: Life would be better if we could retain the named tuple, but IIRC there's code that wants self.urlparts to be a mutable list.

File: linkcheck/checker/httpurl.py [hunk 4]
```diff
@@ -315,7 +303,8 @@ def check_response(self):
 
             if self.url_connection.status_code == 429:
                 self.add_warning(
-                    "Rate limited (Retry-After: %s)" % self.getheader(_("Retry-After")),
+                    "Rate limited (Retry-After: %s)"
+                    % self.headers.get(_("Retry-After")),
```
Comment (mgedmin) [inline linkcheck/checker/httpurl.py:307]: Wait, are we translating the name of an HTTP header????  That would be a bug (the fix for which belongs to a separate PR).

File: linkcheck/checker/internpaturl.py [hunk 5]
```diff
@@ -16,15 +16,18 @@
...
 from . import urlbase, absolute_url
-from .. import strformat, url as urlutil
+from .. import url as urlutil
 
 
 def get_intern_pattern(url):
     """Return intern pattern for given URL. Redirections to the same
     domain with or without "www." prepended are allowed."""
-    parts = strformat.url_unicode_split(url)
+    parts = list(urllib.parse.urlsplit(url))
```
Comment (mgedmin) [inline linkcheck/checker/internpaturl.py:30]: No need for list(), it's a named tuple that supports indexing.

(But for readability we ought to refer to its members by name.  Separate PR material.)

File: linkcheck/checker/httpurl.py (hunk 2 above)
Comment (mgedmin) [reply linkcheck/checker/httpurl.py:114]: Ok, not a bug because this is inside an `if "Content-Length" in self.headers:`, but in that case do not use self.headers.get, use `self.headers['Content-Length']`.

File: linkcheck/plugins/parsepdf.py (hunk 1 above)
Comment (cjmayo) [reply linkcheck/plugins/parsepdf.py:47]: Done.

File: linkcheck/checker/httpurl.py (hunk 2 above)
Comment (cjmayo) [reply linkcheck/checker/httpurl.py:114]: Done.

File: linkcheck/checker/httpurl.py (hunk 4 above)
Comment (cjmayo) [reply linkcheck/checker/httpurl.py:307]: Raised issue #449.

File: linkcheck/checker/internpaturl.py (hunk 5 above)
Comment (cjmayo) [reply linkcheck/checker/internpaturl.py:30]: Removed list()

**Comment by cjmayo** [comment]
Looks like needs a rebase even for Travis which requires the conflicts to be resolved:

So when you create or update a pull request, and Travis CI doesn’t create a build for it, make sure the pull request is mergeable. If it isn’t, rebase it against the upstream branch and resolve any merge conflicts. When you push the fixes up to the pull request, Travis CI will happily build them.

https://docs.travis-ci.com/user/pull-requests/#my-pull-request-isnt-being-built

**Comment by cjmayo** [comment]
I've rebased to unstick Travis and squashed in the review comments.
